lib/models/device.coffee

```diff
@@ -1345,6 +1345,45 @@ getDeviceModel = (deps, opts) ->
 						force: Boolean(options.force)
 		.asCallback(callback)
 
+	###*
+	# @summary Get the supervisor state on a device
+	# @name getSupervisorState
+	# @public
+	# @function
+	# @memberof balena.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.device.getSupervisorState('7cf02a6');
+	#
+	# @example
+	# balena.models.device.getSupervisorState(123);
+	#
+	# @example
+	# balena.models.device.getSupervisorState('7cf02a6',
+	# function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.getSupervisorState = (uuidOrId, callback) ->
+		callback = findCallback(arguments)
+
+		exports.get uuidOrId,
+			$select: 'uuid'
+		.then (device) ->
+			return request.send
+				method: 'POST'
+				url: '/supervisor/v1/device'
+				baseUrl: apiUrl
+				body:
+					uuid: device.uuid
+					method: 'GET'
+		.then ( response ) ->
```

Comment:
This could be
```suggestion
		.get('body')
```
otherwise please format as:
```suggestion
		.then (response) ->
```
But TBH I prefer the `.get()` variant.